src/Microsoft.AspNet.Mvc.Abstractions/ModelBinding/EnumGroupAndName.cs

```diff
@@ -0,0 +1,32 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+namespace Microsoft.AspNet.Mvc.ModelBinding
+{
+    /// <summary>
+    /// An abstraction used when grouping enum values for <see cref="ModelMetadata.EnumGroupedDisplayNamesAndValues"/>
+    /// </summary>
+    public struct EnumGroupAndName
+    {
+        /// <summary>
+        /// Gets the Group name.
+        /// </summary>
+        public string Group { get; }
+
+        /// <summary>
+        /// Gets the name.
+        /// </summary>
+        public string Name { get; }
+
+        /// <summary>
+        /// Initializes a new instance of the EnumGroupAndName structure.
+        /// </summary>
+        /// <param name="group">The group name.</param>
+        /// <param name="name">The name.</param>
+        public EnumGroupAndName(string group, string name)
```

Comment:
- Move constructor to the top.
- Should throw an `ArgumentNullException` if either `group` or `name` are `null`. Code currently calling this is written defensively but using code is not.